src/WebUI/Models/DataState.cs

```diff
@@ -1,13 +1,18 @@
 ﻿using OpenAI.GPT3.ObjectModels.RequestModels;
+using WebUI.Classes;
 
 namespace WebUI.Models;
 
 public class DataState
 {
     public string? OpenAiApiKey { get; set; }
-    public List<ChatMessage> ChatMessages { get; set; } = new();
-    public CancellationTokenSource CancellationTokenSource = new CancellationTokenSource();
+    public ChatLinkedList ChatMessages { get; } = new();
+    public List<ChatLinkedListItem> CurrentMessageThread { get; set; } = new();
+    public CancellationTokenSource CancellationTokenSource { get; set; } = new();
     public OpenAI.GPT3.ObjectModels.Models.Model SelectedGptModel { get; set; } =
         OpenAI.GPT3.ObjectModels.Models.Model.ChatGpt3_5Turbo;
-    public bool UsingByoApiKey { get; set; } = false;
+    public bool UsingByoApiKey { get; set; }
```

Comment:
Is there any reason why this isn't initialised as false anymore?

Reply:
Booleans are initialised to false by default